src/features/wallet/helpers/collectRewards.ts

```diff
@@ -32,46 +33,84 @@ export default function collectRewards(
     .map((item) => item.roundId)
     .filter((value, index, self) => self.indexOf(value) === index);
 
-  uniqueRoundIds.forEach((roundId) => {
-    const mcr = {} as MulticallCollectRewards;
-    const postData = {} as PostRetrieveReward;
-    const pendingRequestData = [] as PendingRequestRetrieveReward[];
+  // Do a multicall request if the user is collecting from multiple rounds.
+  // Otherwise just call the rewards function directly.
+  if (uniqueRoundIds.length > 1) {
+    uniqueRoundIds.forEach((roundId) => {
+      const mcr = {} as MulticallCollectRewards;
+      const postData = {} as PostRetrieveReward;
+      const pendingRequestData = [] as PendingRequestRetrieveReward[];
+
+      data.forEach((datum) => {
```

Comment:
Similar to below, I think this might be a little simpler with a filter + map so you don't have to initialize an array then append to it.

Reply:
Refactored and works great.